CONTRIBUTING.md

```diff
@@ -16,6 +16,111 @@ We are also present on the `#openfeature` channel in the [CNCF slack](https://sl
 
+## How to contribute
+
+All the OpenFeature repositories follow these guidelines.
+
+### Issues
+
+Issues are the heartbeat of open source projects. There are mainly three kinds of issues you can open:
+
+* Bug : If you believe you found a problem in a project, and you want to discuss and get it fixed,
+  creating an issue with the **bug template** is the best way to do so.
+* Documentation: Do want to improve the documentation? did you find a typo? This is the issue for you.
+* Feature: Would you like a new feature to be added to OpenFeature? This is the kind of issue you'll need then.
+  Please do your best at explaining your intent. It's always important that others can understand what you mean in order to discuss.
+  Be open and collaborative in letting others help you get things done!
+* Report a security vulnerability: Did you find a vulnerability? This issue kind is for triaging such problems!
+
+The best way to **get involved** in the project is through issues. You can help in many ways:
+
+* Issues triaging: participating in the discussion and adding details to open issues is always a good thing;
+sometimes issues need to be verified, you could be the one writing a test case to fix a bug!
+* Helping to resolve the issue: you can help in getting it fixed in many ways, more often by opening a pull request.
+
+### Triaging
+
+We need help in categorizing issues. Thus any help is welcome!
+
+When you triage an issue, you:
+
+* assess whether it has merit or not
+
+* quickly close it by correctly answering a question
+
+* point the reporter to a resource or documentation answering the issue
+
+* tag it via labels, projects, or milestones
+
+* take ownership submitting a PR for it, in case you want
+
+#### More about labels
+
+These guidelines are not set in stone and are subject to change.
+
+Each repository has its own set of labels. Here is the current [label set](https://github.com/open-feature/spec/labels) for the spec repository.
+
+### Pull Requests
+
+Thanks for taking the time to make a [pull request](https://help.github.com/articles/about-pull-requests) (hereafter PR).
+
+The PR template is there to guide you through the process of opening it.
+
+Also, feel free to suggest a reviewer with `/cc @theirname`, or to assign an assignee using `/assign @nickname`.
+
+Once your reviewer is happy, they will leave a comment and or approve the pull request.
+
+Your PR will be ready to merge once 2 reviewers have approved it and it has passed all the tests.
+
+### Resolving Conflicts by Rebasing
+
+When submitting a pull request, it's important to make sure that it can be cleanly merged into the upstream repository. In some cases, conflicts may arise when attempting to merge a pull request, such as when changes have been made to the same file or lines of code.
+
+To resolve conflicts, we always use the `git rebase` command rather than git merge. This helps to ensure a clean and linear history of commits.
```

Comment:
We actually do this automatically in all repos I know of. Contributors can add as many commits as they want, and rebase OR merge main as then please, but when the final "merge" is done, it's done as a squash rebase on main automatically, giving us one linear commit.

Reply:
@toddbaert do you suggest removing this part or rewriting?

Reply:
The spirit of the section is really good, which is why we automate it.

I think for reviews, NOT rebasing is actually nicer in most cases, because it doesn't generate new hashes for old commits (rebasing does). Github keeps track of hashes you've "seen" so it can help in reviews that take days or more. 

You might want to mention that we squash/rebase PRs automatically, making the commit message the PR title (which we also audit). This has the net result of keeping a very clean commit history and also makes PRs and reviews pleasant for everyone.